src/howl/core.cljc

```diff
@@ -274,6 +275,32 @@
         (assoc-in state [:block :parse] (second parse))))
     state))
 
+(defn preprocess-literal [literal-block-parse]
+  (map
+   (fn [elem]
+     (if (and (vector? elem) (= :LITERAL (first elem)))
+       [:LITERAL (string/join (drop 1 (butlast elem))) (last elem)]
+       elem))
+   literal-block-parse))
+
+(defn preprocess-block
+  "Given a state map,
+   if it has a :block key with a :parse key,
+   run some post-processing steps on the :parse."
+  [state]
+  (println "COLLAPSING")
+  (if-let [parse (get-in state [:block :parse])]
+    (do (println " DEALING WITH" parse)
+     (case (first parse)
+       :LITERAL_BLOCK  (do
```

Comment:
I prefer to start the `do` on a new line, so that form isn't indented as deeply.